projects/RabbitMQ.Client/client/impl/MainSession.cs

```diff
@@ -44,50 +46,42 @@ namespace RabbitMQ.Client.Impl
     internal sealed class MainSession : Session
     {
         private readonly object _closingLock = new object();
```

Comment:
Could this be optimized out?

Reply:
It could perhaps be replace with an Interlocked.Exchange, but this is called so rarely that I don't think it's necessary. I did add another if check before the lock though which should save us from taking the lock most of the time.

The lock is only taken when we are trying to close the Session so it's rare anywas.

Reply:
We could also lock on the session object or anything else, so we do not require to allocate an additional object for this rare case.

Reply:
Yeah, it's an internal object so we can make sure ourselves noone else is taking that lock. I'll do that instead.